src/porepy/numerics/mixed_dim/assembler.py

```diff
@@ -327,12 +327,22 @@ def assemble_matrix_rhs(
                     # The edge column and row should be assigned to mat_key
                     matrix[mat_key][(ei), (mi, si, ei)] = tmp_mat[(2), (0, 1, 2)]
                     matrix[mat_key][(mi, si), (ei)] = tmp_mat[(0, 1), (2)]
-
                     # Also update the discretization on the master and slave
                     # nodes
                     matrix[mat_key_master][mi, mi] = tmp_mat[0, 0]
                     matrix[mat_key_slave][si, si] = tmp_mat[1, 1]
 
+                    # And add coupling directly between master and slave
+                    # nodes
+                    if matrix[mat_key_master][mi, si] is None:
+                        matrix[mat_key_master][mi, si] = tmp_mat[0, 1]
+                    else:
+                        matrix[mat_key_master][mi, si] += tmp_mat[0, 1]
+
+                    if matrix[mat_key_slave][si, mi] is None:
+                        matrix[mat_key_slave][si, mi] = tmp_mat[1, 0]
+                    else:
+                        matrix[mat_key_slave][si, mi] += tmp_mat[1, 0]
```

Comment:
@keileg What do think about this? This allows for communication between master and slave without going through a mortar variable. I had to do this for my coupled biot-contact mechanics-flow in fractures model to work (I could probably get around it, but this implementation was very convenient). Should we keep this or revert it back to how it was?

Reply:
In principle I am not at all happy with allowing for direct communication, it allows for all kinds of hacks that are not present in the equations. The functional framework requires that a mortar variable is involved in the communication, and I would like to stick with that.

Also, having spent the day working on the equations for contact mechanics, with and without flow, it is not clear to me why you need it - from what I can see, this should all work out with communication via the mortars. Which term is it you want to do this for?

Reply:
It is not strictly needed, but used for convenience. Also, the assembler gives the impression that this is allowed: the coupling discretization gets a 3x3 block matrix and nothing tells you that you can not fill out all the blocks. But I agree, lets remove this from this PR